lib/puppet-lint/plugins/check_classes.rb

```diff
@@ -78,16 +78,15 @@ def check
 # No style guide reference
 PuppetLint.new_check(:class_inherits_from_params_class) do
   def check
-    class_indexes.each do |class_idx|
-      unless class_idx[:inherited_token].nil?
-        if class_idx[:inherited_token].value.end_with? '::params'
-          notify :warning, {
-            :message => 'class inheriting from params class',
-            :line    => class_idx[:inherited_token].line,
-            :column  => class_idx[:inherited_token].column,
-          }
-        end
-      end
+    class_indexes.select { |class_idx|
+      class_idx[:inherited_token] &&
+        class_idx[:inherited_token].value.end_with?('::params')
+    }.each do |class_idx|
+      notify :warning, {
+        :message => 'class inheriting from params class',
+        :line    => class_idx[:inherited_token].line,
+        :column  => class_idx[:inherited_token].column,
+      }
```

Comment:
The second iteration is only trough the offending notes. If performance is the issue, I would look into `lazy`. Anyway, it's not a big difference in the code style, I can agree with your version as well